### ISSUE TITLE
Support for mutable arrays

### ISSUE THREAD

**Issue by simonr0204** [FuelLabs/sway#2457]
Currently, when attempting 

```
let mut my_array: [u64; 3] = [1, 2, 3];
a[1] = 42;
```

The compiler errors with `This expression is not valid on the left hand side of a reassignment`

It would be good to support mutable arrays. This should also be possible in storage, so that:

```
storage {
    my_array: [u64; 3] = [1, 2, 3],
}

//...
storage.my_array[1] = 42
```
also works. (Note: storage arrays would be mutable by definition)

**Comment by mohammadfawaz** [comment]
Yeah mutable arrays on the stack are not supported yet: https://github.com/FuelLabs/sway/issues/428. We have not prioritized that work yet because `Vec` is now available.
Arrays in storage are also not supported yet: https://github.com/FuelLabs/sway/issues/1182.

@simonr0204 are you blocked by this? We are aware of these deficiencies, we just have not prioritized the work yet.

**Comment by simonr0204** [comment]
Not really _blocked_, just seen occasions where a statically sized array is just the right way to model something.

@sezna said

> "Would you mind filing an issue if there isn’t already one? It should not be too hard. Please label it as critical as we should really be able to do this"

So... I did 🙂

**Comment by mohammadfawaz** [comment]
Makes sense! Thanks for filing. I can organize the issues soon :)
